src/workerd/io/tracer.h

```diff
@@ -91,7 +91,8 @@ class PipelineTracer final: public kj::Refcounted, public kj::EnableAddRefToThis
   friend class WorkerTracer;
 
  public:
-  kj::Maybe<kj::Own<tracing::TailStreamWriter>> maybeTailStreamWriter;
+  // tail stream writers for worker stages in the given pipeline.
+  kj::Vector<kj::Own<tracing::TailStreamWriter>> tailStreamWriters;
```

Comment:
I gotta say that this feels odd as a vector, should it really grow and shrink in size throughout the lifetime of the PipelineTracer?
Seems to me like this should be a kj::Array that is passed at construction time.

Reply:
The vector is needed here since tail stream writers are added to the pipeline tracer after it is created, and if there are several workers to be added in the downstream implementation we may not know how many workers need to be added in advance.